src/java/dynamicstructures/SinglyLinkedList.java

```diff
@@ -0,0 +1,76 @@
+package dynamicstructures;
+
+public class SinglyLinkedList {
+    private Node head;
+
+    private class Node {
+        private int data;
+        private Node next;
+
+        private Node(int data) {
+            this.data = data;
+
+        }
+    }
+
+    public void addElement(int data) {
+        Node newNode = new Node(data);
+
+        if (head == null) {
+            head = newNode;
+        } else {
+            Node current = head;
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+    }
+
+
+    public void displayList() {
+        Node current = head;
+
+        while (current != null) {
+            System.out.println(current.data);
+            current = current.next;
+        }
+    }
+
+    public void removeDuplicates() {
+
+        if (head == null) {
+            return;
+        }
+        Node current = head;
+        while (current.next != null) {
+            if (current.data == current.next.data) {
+                current.next = current.next.next;
+            } else {
+                current = current.next;
+            }
+        }
+    }
+
+    public void removeDuplicatesUnsortedList() {
+        if (head == null) {
+            return;
+        }
+        Node current = head;
+
+        while (current != null && current.next != null) {
+            Node fast = current;
+            while(fast.next != null){
+                if(current.data == fast.next.data){
+                    fast.next = fast.next.next;
+                }else{
+                    fast = fast.next;
+                }
+            }
+            current = current.next;
+        }
+    }
```

Comment:
This is good so far, we will try to improve it in next iterations, so far good job!